packages/graphql/test/graphql_client_test.dart

```diff
@@ -128,7 +128,30 @@ void main() {
         expect(nodes[2]['viewerHasStarred'], false);
         return;
       });
-//    test('failed query because of network', {});
+
+      test('failed query because of an exception with null string', () async {
+        final e = Exception();
+        when(mockHttpClient.send(any)).thenAnswer((_) async {
+          throw e;
+        });
+        final Future<QueryResult> r = graphQLClientClient
+            .query(WatchQueryOptions(document: readRepositories));
+
+        await expectLater(r, throwsA(e));
+        return;
+      });
+
+      test('failed query because of an exception with empty string', () async {
+        final e = Exception('');
+        when(mockHttpClient.send(any)).thenAnswer((_) async {
+          throw e;
+        });
```

Comment:
Nit: add a new line between the mock and the test execution. This will make it easier to read.